`Type VAL (+TypeVariableImpl is not supported yet.`

Hey,

I have another issue. When running your plugin I'm getting this error: `Type VAL (+TypeVariableImpl is not supported yet.`. I don't know what is causing it, maybe Lombok?

```
Caused by: java.lang.RuntimeException: Type VAL (+TypeVariableImpl is not supported yet.
    at io.github.kbuntrock.model.DataObject.<init> (DataObject.java:141)
    at io.github.kbuntrock.TagLibrary.inspectObject (TagLibrary.java:113)
    at io.github.kbuntrock.TagLibrary.exploreDataObject (TagLibrary.java:87)
    at io.github.kbuntrock.TagLibrary.inspectObject (TagLibrary.java:114)
    at io.github.kbuntrock.TagLibrary.exploreDataObject (TagLibrary.java:77)
    at io.github.kbuntrock.TagLibrary.exploreDataObject (TagLibrary.java:84)
```

https://github.com/kbuntrock/openapi-maven-plugin/blob/1184f467bd18bf3d8104ca30db0abe9dc903d2ad/openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/DataObject.java#L141

Ps In error message in `DataObject.java#L140` you are missing closing `)` 😉

--- a/openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/DataObject.java
+++ b/openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/DataObject.java
@@ -139,7 +139,7 @@ public class DataObject {
 			} else {
 
 				//	TypeToken.of(this.getJavaClass()).resolveType(genericType).getType();
-				
+
 				throw new RuntimeException(
 					"Type " + originalType.getTypeName() + " (+" + originalType.getClass().getSimpleName() + ") is not supported yet.");
 			}
@@ -323,27 +323,26 @@ public class DataObject {
 				}
 
 			}
-		} else {
-			// A not generic object type does not mean we are free from genericity ...
-			if(genericType instanceof ParameterizedType) {
-				// Here we handle "Class<? extends XXX> which can not be substituted locally.
-				final ParameterizedType parameterizedType = (ParameterizedType) genericType;
-				if(parameterizedType.getRawType() == Class.class && parameterizedType.getActualTypeArguments().length == 1
-					&& parameterizedType.getActualTypeArguments()[0] instanceof WildcardType) {
-					final WildcardType wt = (WildcardType) parameterizedType.getActualTypeArguments()[0];
-					if(wt.getLowerBounds().length == 0 && wt.getUpperBounds().length == 1) {
-						// Return "XXX" as the only type we can determine for this object.
-						// The implementation surely will be a child of this type but we can't guess it.
-						return wt.getUpperBounds()[0];
-					}
-				}
-			} else if(genericType instanceof TypeVariable) {
-
-				// We are in presence of a generic type variable not coming from the outside. Might be coming from generic typing at a parent level.
-				return TypeToken.of(this.getJavaClass()).resolveType(genericType).getType();
-			}
-
 		}
+		// A not generic object type does not mean we are free from genericity ...
+		if(genericType instanceof ParameterizedType) {
+			// Here we handle "Class<? extends XXX> which can not be substituted locally.
+			final ParameterizedType parameterizedType = (ParameterizedType) genericType;
+			if(parameterizedType.getRawType() == Class.class && parameterizedType.getActualTypeArguments().length == 1
+				&& parameterizedType.getActualTypeArguments()[0] instanceof WildcardType) {
+				final WildcardType wt = (WildcardType) parameterizedType.getActualTypeArguments()[0];
+				if(wt.getLowerBounds().length == 0 && wt.getUpperBounds().length == 1) {
+					// Return "XXX" as the only type we can determine for this object.
+					// The implementation surely will be a child of this type but we can't guess it.
+					return wt.getUpperBounds()[0];
+				}
+			}
+		} else if(genericType instanceof TypeVariable) {
+
+			// We are in presence of a generic type variable not coming from the outside. Might be coming from generic typing at a parent level.
+			return TypeToken.of(this.getJavaClass()).resolveType(genericType).getType();
+		}
+		
 		return genericType;
 	}
 

--- a/openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/DataObject.java
+++ b/openapi-maven-plugin/src/main/java/io/github/kbuntrock/model/DataObject.java
@@ -137,8 +137,11 @@ public class DataObject {
 				// Anything simplier ...
 				javaClass = (Class<?>) type;
 			} else {
+
+				//	TypeToken.of(this.getJavaClass()).resolveType(genericType).getType();
+				
 				throw new RuntimeException(
-					"Type " + originalType.getTypeName() + " (+" + originalType.getClass().getSimpleName() + " is not supported yet.");
+					"Type " + originalType.getTypeName() + " (+" + originalType.getClass().getSimpleName() + ") is not supported yet.");
 			}
 
 			this.openApiType = OpenApiDataType.fromJavaClass(javaClass);

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java
@@ -37,6 +37,7 @@ import io.github.kbuntrock.resources.endpoint.generic.GenericityTestThree;
 import io.github.kbuntrock.resources.endpoint.generic.GenericityTestTwelve;
 import io.github.kbuntrock.resources.endpoint.generic.GenericityTestTwo;
 import io.github.kbuntrock.resources.endpoint.generic.Issue89;
+import io.github.kbuntrock.resources.endpoint.generic.Issue95;
 import io.github.kbuntrock.resources.endpoint.ignore.JsonIgnoreController;
 import io.github.kbuntrock.resources.endpoint.interfacedto.InterfaceController;
 import io.github.kbuntrock.resources.endpoint.map.MapController;
@@ -226,6 +227,19 @@ public class SpringClassAnalyserTest extends AbstractTest {
 
 		final List<File> generated = mojo.documentProject();
 		checkGenerationResult("ut/SpringClassAnalyserTest/issue_89.yml", generated.get(0));
+	}
+
+	@Test
+	public void issue_95() throws MojoFailureException, IOException, MojoExecutionException {
+
+		final DocumentationMojo mojo = createBasicMojo(Issue95.class.getCanonicalName());
+		final JavadocConfiguration javadocConfig = new JavadocConfiguration();
+//		javadocConfig.setScanLocations(Arrays.asList("src/test/java/io/github/kbuntrock/resources/endpoint/generic",
+//			"src/test/java/io/github/kbuntrock/resources/dto/genericity/issue89"));
+		mojo.setJavadocConfiguration(javadocConfig);
+
+		final List<File> generated = mojo.documentProject();
+		checkGenerationResult("ut/SpringClassAnalyserTest/issue_95.yml", generated.get(0));
 	}
 
 	@Test

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/SpringClassAnalyserTest.java
@@ -234,8 +234,8 @@ public class SpringClassAnalyserTest extends AbstractTest {
 
 		final DocumentationMojo mojo = createBasicMojo(Issue95.class.getCanonicalName());
 		final JavadocConfiguration javadocConfig = new JavadocConfiguration();
-//		javadocConfig.setScanLocations(Arrays.asList("src/test/java/io/github/kbuntrock/resources/endpoint/generic",
-//			"src/test/java/io/github/kbuntrock/resources/dto/genericity/issue89"));
+		javadocConfig.setScanLocations(Arrays.asList("src/test/java/io/github/kbuntrock/resources/endpoint/generic",
+			"src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95"));
 		mojo.setJavadocConfiguration(javadocConfig);
 
 		final List<File> generated = mojo.documentProject();
@@ -620,7 +620,7 @@ public class SpringClassAnalyserTest extends AbstractTest {
 		blackList.add(".*\\.GenericityTestS.*$");
 		blackList.add(".*\\.GenericityTestE.*$");
 		blackList.add(".*\\.GenericityTestN.*$");
-		blackList.add(".*\\.Issue89.*$");
+		blackList.add(".*\\.Issue.*$");
 		blackList.add(".*\\.ActionResource");
 		blackList.add(".*\\.GenericDataController");
 		mojo.getApiConfiguration().setBlackList(blackList);

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/BaseValue.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/BaseValue.java
@@ -2,9 +2,25 @@ package io.github.kbuntrock.resources.dto.genericity.issue95;
 
 import java.util.Optional;
 
+/**
+ * This is the BaseValue declaration
+ *
+ * @param <T>   a first generic type
+ * @param <VAL> another generic type
+ */
 public interface BaseValue<T, VAL> {
 
+	/**
+	 * Get the value!
+	 *
+	 * @return the returned value
+	 */
 	VAL getValue();
 
+	/**
+	 * Get the unit!
+	 *
+	 * @return the optional returned unit
+	 */
 	Optional<T> getUnit();
 }

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/BaseValue.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/BaseValue.java
@@ -1,0 +1,10 @@
+package io.github.kbuntrock.resources.dto.genericity.issue95;
+
+import java.util.Optional;
+
+public interface BaseValue<T, VAL> {
+
+	VAL getValue();
+
+	Optional<T> getUnit();
+}

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorDto.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorDto.java
@@ -1,0 +1,6 @@
+package io.github.kbuntrock.resources.dto.genericity.issue95;
+
+public interface ErrorDto {
+
+	ErrorFoo<String> getFoo();
+}

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorDto.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorDto.java
@@ -1,6 +1,14 @@
 package io.github.kbuntrock.resources.dto.genericity.issue95;
 
+/**
+ * This is the ErrorDto declaration
+ */
 public interface ErrorDto {
 
+	/**
+	 * Get the foo
+	 *
+	 * @return a string type ErrorFoo
+	 */
 	ErrorFoo<String> getFoo();
 }

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorFoo.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorFoo.java
@@ -1,0 +1,7 @@
+package io.github.kbuntrock.resources.dto.genericity.issue95;
+
+import java.math.BigDecimal;
+
+public interface ErrorFoo<T> extends BaseValue<T, BigDecimal> {
+
+}

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorFoo.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/dto/genericity/issue95/ErrorFoo.java
@@ -2,6 +2,11 @@ package io.github.kbuntrock.resources.dto.genericity.issue95;
 
 import java.math.BigDecimal;
 
+/**
+ * This is the ErrorFoo declaration
+ *
+ * @param <T> a generic type
+ */
 public interface ErrorFoo<T> extends BaseValue<T, BigDecimal> {
 
 }

--- a/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/endpoint/generic/Issue95.java
+++ b/openapi-maven-plugin/src/test/java/io/github/kbuntrock/resources/endpoint/generic/Issue95.java
@@ -1,0 +1,16 @@
+package io.github.kbuntrock.resources.endpoint.generic;
+
+import io.github.kbuntrock.resources.dto.genericity.issue95.ErrorDto;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * @author Kévin Buntrock
+ */
+@RequestMapping("/issue95")
+public interface Issue95 {
+
+	@PutMapping(path = "/error")
+	ErrorDto error();
+
+}

--- a/openapi-maven-plugin/src/test/resources/ut/SpringClassAnalyserTest/issue_95.yml
+++ b/openapi-maven-plugin/src/test/resources/ut/SpringClassAnalyserTest/issue_95.yml
@@ -1,0 +1,42 @@
+---
+openapi: 3.0.3
+info:
+  title: My Project
+  version: 10.5.36
+servers:
+  - url: ""
+tags:
+  - name: Issue95
+paths:
+  /issue95/error:
+    put:
+      tags:
+        - Issue95
+      operationId: error
+      responses:
+        200:
+          description: successful operation
+          content:
+            '*/*':
+              schema:
+                $ref: '#/components/schemas/ErrorDto'
+components:
+  schemas:
+    ErrorDto:
+      description: This is the ErrorDto declaration
+      type: object
+      properties:
+        foo:
+          description: Get the foo
+          type: object
+          properties:
+            unit:
+              description: Get the unit!
+              type: object
+              properties:
+                value:
+                  type: string
+            value:
+              description: Get the value!
+              type: number
+              format: double

